examples/addons/nonblocking.py

```diff
@@ -1,17 +1,13 @@
 """
-Make events hooks non-blocking.
-
-When event hooks are decorated with @concurrent, they will be run in their own thread, freeing the main event loop.
-Please note that this generally opens the door to race conditions and decreases performance if not required.
+Make events hooks non-blocking using async.
 """
-import time
+import asyncio
 
-from mitmproxy.script import concurrent
+from mitmproxy import ctx
 
 
-@concurrent  # Remove this and see what happens
-def request(flow):
-    # This is ugly in mitmproxy's UI, but you don't want to use mitmproxy.ctx.log from a different thread.
-    print(f"handle request: {flow.request.host}{flow.request.path}")
-    time.sleep(5)
-    print(f"start  request: {flow.request.host}{flow.request.path}")
+# Change this to a sync hook using time.sleep and see what happens
```

Comment:
Nit: Can we add a commented sync version below and have a `# use this version instead and see what happens` comment? I think "changing to a sync hook" is not trivial to do for newcomers. :)

Reply:
Add: We also want to keep `@concurrent` (see main comment), so it make sense to demonstrate 1) async 2) `@concurrent` and 3) the bad blocking sleep() in here. :)